pkg/agent/kubeagent.go

```diff
@@ -126,3 +169,261 @@ func (k *KubeAgent) GetFactories(_ context.Context) (otelcol.Factories, error) {
+	}
+
+	// Read response body
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		c.logger.Error("failed to reas response body", zap.Error(err))
+		return err
+	}
+
+	// Unmarshal JSON response into ApiResponse struct
+	var apiResponse apiResponseForYAML
+	if err := json.Unmarshal(body, &apiResponse); err != nil {
+		c.logger.Error("failed to unmarshal api response", zap.Error(err))
+		return err
+	}
+
+	// Verify API Response
+	if !apiResponse.Status {
+		c.logger.Error("failure status from api response for ingestion rules", zap.Bool("status", apiResponse.Status))
+		return ErrInvalidResponse
+	}
+
+	var apiYAMLConfig map[string]interface{}
+	if len(apiResponse.Config.DaemonSet) == 0 && len(apiResponse.Config.Deployment) == 0 {
+		c.logger.Error("failed to get valid response",
+			zap.Int("config docker len", len(apiResponse.Config.Docker)),
+			zap.Int("config no docker len", len(apiResponse.Config.NoDocker)))
+		return ErrInvalidResponse
+	}
+
+	// apiYAMLConfig = apiResponse.Config.NoDocker
+	if componentType == DaemonSet {
+		apiYAMLConfig = apiResponse.Config.DaemonSet
+	} else {
+		apiYAMLConfig = apiResponse.Config.Deployment
+	}
+
+	yamlData, err := yaml.Marshal(apiYAMLConfig)
+	if err != nil {
+		c.logger.Error("failed to marshal api data", zap.Error(err))
+		return err
+	}
+
+	switch componentType {
+	case DaemonSet:
+		// Retrieve the existing ConfigMap
+		existingDaemonsetConfigMap, err := c.Clientset.CoreV1().ConfigMaps(c.AgentNamespace).Get(context.Background(), c.DaemonsetConfigMap, metav1.GetOptions{})
+		if err != nil {
+			c.logger.Error("Error getting ConfigMap: %v\n" + err.Error())
+		}
+
+		// Modify the content of the ConfigMap
+		existingDaemonsetConfigMap.Data["otel-config"] = string(yamlData)
+
+		// Update the ConfigMap
+		updatedConfigMap, err := c.Clientset.CoreV1().ConfigMaps(c.AgentNamespace).Update(context.Background(), existingDaemonsetConfigMap, metav1.UpdateOptions{})
+		if err != nil {
+			c.logger.Error("Error updating ConfigMap ", zap.String("error", err.Error()))
+		}
+
+		c.logger.Info("ConfigMap updated successfully ", zap.String("configmap", updatedConfigMap.Name))
+	case Deployment:
+		// Retrieve the existing ConfigMap
+		existingDeploymentConfigMap, err := c.Clientset.CoreV1().ConfigMaps(c.AgentNamespace).Get(context.Background(), c.DeploymentConfigMap, metav1.GetOptions{})
+		if err != nil {
+			c.logger.Error("Error getting ConfigMap ", zap.String("error", err.Error()))
+		}
+
+		// Modify the content of the ConfigMap
+		existingDeploymentConfigMap.Data["otel-config"] = string(yamlData)
+
+		// Update the ConfigMap
+		updatedDeploymentConfigMap, err := c.Clientset.CoreV1().ConfigMaps(c.AgentNamespace).Update(context.Background(), existingDeploymentConfigMap, metav1.UpdateOptions{})
+		if err != nil {
+			c.logger.Error("Error updating ConfigMap", zap.String("error", err.Error()))
+		}
+
+		c.logger.Info("ConfigMap updated successfully", zap.String("configmap", updatedDeploymentConfigMap.Name))
```

Comment:
We will log `ConfigMap updated successfully` even if there is a  failure at line 421